.circleci/config.yml

```diff
@@ -54,18 +55,98 @@ jobs:
           echo 'export SKIP_METRICS=y' >> $BASH_ENV
           echo 'export STAGE_TIMEOUT_MIN=30' >> $BASH_ENV
           echo 'export NUM_OF_RETRIES=2' >> $BASH_ENV
-          echo 'export TEST_CONFIG=test/acse-conf/acse-feature-validation.json' >> $BASH_ENV
-          echo 'export SUBSCRIPTION_ID=${SUBSCRIPTION_ID_FEATURE_VALIDATION}' >> $BASH_ENV
-          echo 'export SERVICE_PRINCIPAL_CLIENT_ID=${SERVICE_PRINCIPAL_CLIENT_ID_FEATURE_VALIDATION}' >> $BASH_ENV
-          echo 'export SERVICE_PRINCIPAL_CLIENT_SECRET=${SERVICE_PRINCIPAL_CLIENT_SECRET_FEATURE_VALIDATION}' >> $BASH_ENV
-          echo 'export CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID=${CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID_FEATURE_VALIDATION}' >> $BASH_ENV
-          echo 'export CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET=${CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET_FEATURE_VALIDATION}' >> $BASH_ENV
+          echo 'export SUBSCRIPTION_ID=${SUBSCRIPTION_ID_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export SERVICE_PRINCIPAL_CLIENT_ID=${SERVICE_PRINCIPAL_CLIENT_ID_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export SERVICE_PRINCIPAL_CLIENT_SECRET=${SERVICE_PRINCIPAL_CLIENT_SECRET_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID=${SERVICE_PRINCIPAL_CLIENT_ID_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET=${SERVICE_PRINCIPAL_CLIENT_SECRET_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export TEST_CONFIG=test/e2e/kubernetes-deployments.json' >> $BASH_ENV
       - run:
-          name: Feature validation tests
+          name: Kubernetes feature validation tests
+          command: make bootstrap build test-e2e
+      - store_artifacts:
+          path: /go/src/github.com/Azure/acs-engine/_logs
+  master-swarm-e2e:
+    working_directory: /go/src/github.com/Azure/acs-engine
+    no_output_timeout: "30m"
+    docker:
+      - image: microsoft/acs-engine:latest
```

Comment:
What's the purpose of this image here in this part of the CI run? Does it hold the binary we are testing? If so when did it get built?